### PR TITLE
[DYNAREC] Fix x87cache issues

### PR DESCRIPTION
**Pull request by ksco**
Reported by @fan-wenjie

### CODE REVIEW

**Comment by ksco** [comment]
@ptitSeb Seems working, but I can't test real programs right now.

**Comment by ptitSeb** [comment]
Looks ok. I would have created new macro for PUSH/PUSHEMPTY/POP for easier changes, but this one works too.

I can wait if you can test a bit later.

**Comment by ksco** [comment]
> I can wait if you can test a bit later.

Are you able to test it for arm64? Like running some games (UT2004 maybe?) or WOW64 apps that use x87, I want to make sure it doesn't break things that are already working. I'll do some tests for rv64 later today.

> I would have created new macro for PUSH/PUSHEMPTY/POP

Yeah, this looks better, but we'll only need to create new macros for PUSH/PUSHEMPTY, POP empty is harder to detect as we assume the x87cache is empty at the entry of a block IIUC.

**Comment by ptitSeb** [comment]
> 
> > I would have created new macro for PUSH/PUSHEMPTY/POP
> 
> Yeah, this looks better, but we'll only need to create new macros for PUSH/PUSHEMPTY, POP empty is harder to detect as we assume the x87cache is empty at the entry of a block IIUC.

Pop empty is the same: you need to test if the stack is -8 (because it can be negative, if you start a block and just pop po pop...)

File: src/dynarec/arm64/dynarec_arm64_helper.c [hunk 1]
```diff
@@ -974,6 +974,16 @@ void x87_do_pop(dynarec_arm_t* dyn, int ninst, int s1)
         }
 }
 
+int x87_cachecount(dynarec_arm_t* dyn, int ninst)
```
Comment (ksco) [inline src/dynarec/arm64/dynarec_arm64_helper.c:977]: Ah, I see, we already have `dyn->n.stack` for this, so this function is unnecessary then...

**Comment by ksco** [comment]
Ok, I've tested some games on rv64, seems all good ;)

**Comment by ptitSeb** [comment]
Ok, I'll push then.
